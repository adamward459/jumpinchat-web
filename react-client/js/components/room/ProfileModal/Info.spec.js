import React from 'react';
import { shallow } from 'enzyme';
import ProfileInfo from './Info.react';

describe('ProfileInfo', () => {
  let props;

  beforeEach(() => {
    props = {
      closeModal: jest.fn(),
      ignoreListItem: null,
      profile: {
        userListId: 'foo',
      },
      roomOwner: null,
      user: {
        isAdmin: false,
        operatorPermissions: null,
        user_id: null,
      },
    };
  });

  it('should explain guest user by default', () => {
    props.profile.handle = 'foo';
    const wrapper = shallow(<ProfileInfo {...props} />);
    expect(wrapper.getElement()).toMatchSnapshot();
  });

  it('should show join info if user has join and last seen data', () => {
    props.profile.username = 'foo';
    props.profile.userType = 'awesome';
    props.profile.handle = 'bar';
    props.profile.joinDate = '2019-01-01T00:00:00.000Z';
    props.profile.lastSeen = '2019-01-01T00:00:00.000Z';

    const wrapper = shallow(<ProfileInfo {...props} />);
    expect(wrapper.getElement()).toMatchSnapshot();
  });

  it('should show user type and username', () => {
    props.profile.username = 'foo';
    props.profile.userType = 'awesome';
    props.profile.handle = 'bar';
    const wrapper = shallow(<ProfileInfo {...props} />);
    expect(wrapper.getElement()).toMatchSnapshot();
  });
});
