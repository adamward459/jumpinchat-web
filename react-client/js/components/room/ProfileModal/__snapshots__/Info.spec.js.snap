// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileInfo should explain guest user by default 1`] = `
<div
  className="profile__Info"
>
  foo
   is a 
  <strong>
    guest user
  </strong>
</div>
`;

exports[`ProfileInfo should show join info if user has join and last seen data 1`] = `
<div
  className="profile__Info"
>
  foo
   is a 
  <strong>
    awesome
  </strong>
  <React.Fragment>
     
    who joined 
    01/01/2019
     and was last seen 
    01/01/2019
  </React.Fragment>
</div>
`;

exports[`ProfileInfo should show user type and username 1`] = `
<div
  className="profile__Info"
>
  foo
   is a 
  <strong>
    awesome
  </strong>
</div>
`;
